doc/api/deprecations.md

```diff
@@ -0,0 +1,574 @@
+Please specify a digest.
+
+<a id="DEP0010"></a>
+### DEP0010: crypto.createCredentials
+
+Type: Runtime
+
+The [`crypto.createCredentials()`][] API is deprecated. Please use
+[`tls.createSecureContext()`][] instead.
+
+<a id="DEP0011"></a>
+### DEP0011: crypto.Credentials
+
+Type: Runtime
+
+The `crypto.Credentials` class is deprecated. Please use [`tls.SecureContext`][]
+instead.
+
+<a id="DEP0012"></a>
+### DEP0012: Domain.dispose
+
+Type: Runtime
+
+[`Domain.dispose()`][] is deprecated. Recover from failed I/O actions
+explicitly via error event handlers set on the domain instead.
+
+<a id="DEP0013"></a>
+### DEP0013: fs async function without callback
+
+Type: Runtime
+
+Calling an asynchronous function without callback is deprecated.
+
+<a id="DEP0014"></a>
+### DEP0014: fs.read legacy String interface
+
+Type: Runtime
+
+The [`fs.read()`][] legacy String interface is deprecated. Use the Buffer API as
+mentioned in the documentation instead.
+
+<a id="DEP0015"></a>
+### DEP0015: fs.readSync legacy String interface
+
+Type: Runtime
+
+The [`fs.readSync()`][] legacy String interface is deprecated. Use the Buffer
+API as mentioned in the documentation instead.
+
+<a id="DEP0016"></a>
+### DEP0016: GLOBAL/root
+
+Type: Runtime
+
+The `GLOBAL` and `root` aliases for the `global` property have been deprecated
+and should no longer be used.
+
+<a id="DEP0017"></a>
+### DEP0017: Intl.v8BreakIterator
+
+Type: Runtime
+
+The `Intl.v8BreakIterator` is deprecated and will be removed or replaced soon.
+
+<a id="DEP0018"></a>
+### DEP0018: Unhandled promise rejections
+
+Type: Runtime
+
+Unhandled promise rejections are deprecated. In the future, promise rejections
+that are not handled will terminate the Node.js process with a non-zero exit
+code.
+
+<a id="DEP0019"></a>
+### DEP0019: require('.') resolved outside directory
+
+Type: Runtime
+
+In certain cases, `require('.')` may resolve outside the package directory.
+This behavior is deprecated and will be removed in a future major Node.js
+release.
+
+<a id="DEP0020"></a>
+### DEP0020: Server.connections
+
+Type: Runtime
+
+The [`Server.connections`][] property is deprecated. Please use the
+[`Server.getConnections()`][] method instead.
+
+<a id="DEP0021"></a>
+### DEP0021: Server.listenFD
+
+Type: Runtime
+
+The `Server.listenFD()` method is deprecated. Please use
+[`Server.listen({fd: <number>})`][] instead.
+
+<a id="DEP0022"></a>
+### DEP0022: os.tmpDir()
+
+Type: Runtime
+
+The `os.tmpDir()` API is deprecated. Please use [`os.tmpdir()`][] instead.
+
+<a id="DEP0023"></a>
+### DEP0023: os.getNetworkInterfaces()
+
+Type: Runtime
+
+The `os.getNetworkInterfaces()` method is deprecated. Please use the
+[`os.networkInterfaces`][] property instead.
+
+<a id="DEP0024"></a>
+### DEP0024: REPLServer.prototype.convertToContext()
+
+Type: Runtime
+
+The `REPLServer.prototype.convertToContext()` API is deprecated and should
+not be used.
```

Comment:
While I agree that the docs can be better, I want to avoid going down the rabbit hole with each of the deprecation descriptions in this PR. Those can be fixed up in subsequent PRs